src/framework/entity.js

```diff
@@ -379,6 +379,36 @@ class Entity extends GraphNode {
         });
     }
 
+    /**
+     * Search the entity and all of its descendants for the first script instance of specified type.
+     *
+     * @param {string|Class<import('./script/script-type.js').ScriptType>} nameOrType - The name or type of {@link ScriptType}.
+     * @returns {import('./script/script-type.js').ScriptType} A script instance of specified type, if the entity or any of its descendants
```

Comment:
add `|undefined` to the return type please. (we should fix the findComponent too)